src/components/inputs/money-input/money-input.spec.js

```diff
@@ -84,25 +85,42 @@ describe('MoneyInput.getAmountInputId', () => {
 describe('MoneyInput.convertToMoneyValue', () => {
   describe('when there is no currency code', () => {
     it('should return an invalid object', () => {
-      expect(MoneyInput.convertToMoneyValue({ amount: '1' })).toEqual(null);
+      expect(MoneyInput.convertToMoneyValue({ amount: '1' }, 'en')).toEqual(
+        null
+      );
     });
   });
   describe('when an unknown currency is used', () => {
     it('should return an invalid object', () => {
       expect(
-        MoneyInput.convertToMoneyValue({ currencyCode: 'foo', amount: '1' })
+        MoneyInput.convertToMoneyValue(
+          { currencyCode: 'foo', amount: '1' },
+          'en'
+        )
       ).toEqual(null);
     });
   });
+  describe('when an unknown currency is used2', () => {
+    it('should return an invalid object', () => {
+      expect(() =>
+        MoneyInput.convertToMoneyValue({ currencyCode: 'JPY', amount: '1' })
+      ).toThrow(
+        'A locale must be provided when currency has no fraction digits (JPY)'
+      );
+    });
+  });
 
   describe('when no amount is present', () => {
     it('should return an invalid object', () => {
       expect(
-        MoneyInput.convertToMoneyValue({ currencyCode: 'EUR', amount: '' })
+        MoneyInput.convertToMoneyValue(
+          { currencyCode: 'EUR', amount: '' },
+          'en'
```

Comment:
is this right?

Reply:
yes, but the default locales are not needed though